### PR TITLE
feat(ecau): extract Soundcloud backdrop images

### PR DESCRIPTION
**Pull request by ROpdebee**
While fixing #526 I noticed that the Soundcloud tracks in that set have backdrop images, so let's extract those too.
Also included is a fix for #526. Not sure if this problem has always existed or was caused by a Soundcloud change, because we didn't have tests for sets with >5 tracks.

### CODE REVIEW

File: src/mb_enhanced_cover_art_uploads/providers/soundcloud.ts [hunk 1]
```diff
@@ -120,16 +150,77 @@ export class SoundcloudProvider extends ProviderWithTrackImages {
         // Don't bother extracting track covers if they won't be used anyway
         if (onlyFront) return covers;
 
-        const trackCovers = filterNonNull(metadata.data.tracks
+        // Soundcloud only loads data for the first 5 tracks at first, need to
+        // load the rest of the tracks later.
```
Comment (kellnerd) [inline src/mb_enhanced_cover_art_uploads/providers/soundcloud.ts:154]: I assume later = now, i.e. in the next line?

File: src/mb_enhanced_cover_art_uploads/providers/base.ts [hunk 2]
```diff
@@ -116,6 +116,7 @@ export abstract class CoverArtProvider {
 export interface ParsedTrackImage {
     url: string;
     trackNumber?: string;
+    customComment?: [string, string];  // singular, plural
```
Comment (kellnerd) [inline src/mb_enhanced_cover_art_uploads/providers/base.ts:119]:
```suggestion
    customCommentPrefix?: [string, string];  // singular, plural
```
Makes it clearer whether we expect a comment prefix or a full comment which will be used verbatim.

Comment (ROpdebee) [reply src/mb_enhanced_cover_art_uploads/providers/base.ts]: Good point. I'm sure you'll find loads of other poorly-named variables/properties in the changes over the last couple of months 😅

**Review by kellnerd — APPROVED** [review]
Looking good, but I did not test 😇

**Pull request merged**